### PR TITLE
Do not look for override method if there is no form/parseable data

### PR DESCRIPTION
**Pull request by jcsjcs**
See discussion at https://github.com/jeremyevans/roda/discussions/291

### CODE REVIEW

**Comment by jeremyevans** [comment]
Change looks good, but this needs tests before it can be committed.

**Comment by jcsjcs** [comment]
Sorry, test added.

**Review by jeremyevans — APPROVED** [review]

File: lib/rack/method_override.rb [hunk 1]
```diff
@@ -46,7 +46,7 @@ def allowed_methods
     end
 
     def method_override_param(req)
-      req.POST[METHOD_OVERRIDE_PARAM_KEY]
+      req.POST[METHOD_OVERRIDE_PARAM_KEY] if req.form_data? || req.parseable_data?
```
Comment (ioquatix) [inline lib/rack/method_override.rb:49]: I will look myself, but if you already know you can help me, there is no significant performance cost of calling these methods right? (i.e. it's not parsing the form body etc).

Comment (jeremyevans) [reply lib/rack/method_override.rb:49]: The methods just check the request method and headers, they don't deal with the body.

File: test/spec_method_override.rb [hunk 2]
```diff
@@ -112,4 +112,15 @@ def app
 
     env["REQUEST_METHOD"].must_equal "POST"
   end
+
+  it "not set form input when the content type is JSON" do
+    env = Rack::MockRequest.env_for("/",
+            "CONTENT_TYPE" => "application/json",
+            method: "POST",
+            input: '{"_method":"options"}')
+    app.call env
```
Comment (ioquatix) [inline test/spec_method_override.rb:121]:
```suggestion
    env = Rack::MockRequest.env_for("/",
      "CONTENT_TYPE" => "application/json",
      method: "POST",
      input: '{"_method":"options"}')
    app.call env
```

**Pull request merged**